src/mscorlib/System.Private.CoreLib.csproj

```diff
@@ -33,7 +33,7 @@
     <GenerateTargetFrameworkAttribute>false</GenerateTargetFrameworkAttribute>
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
-    <DefineConstants>$(DefineConstants);CORECLR;_USE_NLS_PLUS_TABLE;RESOURCE_SATELLITE_CONFIG;CODE_ANALYSIS_BASELINE</DefineConstants>
+    <DefineConstants>$(DefineConstants);CORECLR;_USE_NLS_PLUS_TABLE;RESOURCE_SATELLITE_CONFIG;CODE_ANALYSIS_BASELINE;netcoreapp</DefineConstants>
```

Comment:
See https://github.com/dotnet/coreclr/pull/16733#discussion_r172253885

Basically, this code is compiled here for `netcoreap`, and in CoreFX for `netfx` and `netstandard`.  So just `!netstandard` isn't going to be enough.

Reply:
Thanks for linking to the previous discussion. I had missed it. Can we instead define netcoreapp on the corefx side instead? Having an `#if netcoreapp` within coreclr seems weird.

Reply:
IMO it only seems weird until you realize that this file is "shared" outside of coreclr.

The other way to make it work is to say:

`#if !netfx and !netstandard`

And then define `netfx` and `netstandard` in CoreFX.

But then it would seem just as weird that you'd have `#if !netfx`.  This is coreclr - we shouldn't have any "netfx" code in it....

Reply:
> But then it would seem just as weird that you'd have `#if !netfx`.  This is coreclr - we shouldn't have any "netfx" code in it....

Agreed. I retract my concern :)